### PR TITLE
snapcraft.yaml: update to work with modern snapcraft

### PR DESCRIPTION
**Pull request by mvo5**
spread is currently not build with snapcraft 5.0 (the default).
This commit fixes this by moving to `base: core20`. 

Note that this also requires moving to `go.mod` hand hence it includes #129.

### CODE REVIEW

File: snapcraft.yaml
Comment (sergiusens) [inline snapcraft.yaml:18]: adding an entry for personal-files to be able to easily use the google backend would be nice.

Comment (mvo5) [reply snapcraft.yaml:18]: Excellent suggestion, I think this would ideally be a followup?

**Review by sergiusens — APPROVED** [review]
I would update the version to something relevant (not blocking on it) and consider another PR to move to core20. This works for now.

File: snapcraft.yaml
Comment (mvo5) [reply snapcraft.yaml:2]: I will do a followup with this change and also the move to core22. This PR got (soft) approval by Gustavo as is so I don't want to add more potentially controversial things just now :)

Comment (mvo5) [reply snapcraft.yaml:2]: I addressed the version in https://github.com/snapcore/spread/pull/161 - please let me know if that matches what you had in mind and I'm happy to tweak as needed.

**Pull request merged**
